.github/workflows/tests.yml

```diff
@@ -19,37 +23,39 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: ["2.12", "2.17"]
+        version: ["2.12.4", "2.19.0"]
```

Comment:
Why do you start at 2.12 and not before? Why is it not relevant anymore to test other versions between 12 and 17? And lastly, why do we need to test multiple versions since they are not breaking?

Reply:
> Why 2.12?

Dart 2.12 added support for [sound null safety](https://dart.dev/null-safety). When you opt into null safety, types in your code are non-nullable by default, meaning that variables can’t contain null unless you say they can. Even the dart image tags available in the docker hub start from 2.12.

> Why not test other versions between 12 and 17?

That's a good question, but in fact, version 2.17 was the latest available when the changes were made, and 12 because they are not compatible with older versions. The in-between versions should be compatible. 

> Why do we need to test multiple versions since they are not breaking?

I'm testing the minimum required to work, and the last available, just in case the Dart team introduces a new incompatible change like in the 2.12.